src/commands/baseTypeConverters.ts

```diff
@@ -85,13 +85,13 @@ export const baseTypeConverters = {
     return channel;
   },
 
-  textChannel(value: string, { message }: CommandContext<AnyPluginData<any>>): TextChannel {
+  textChannel(value: string, { message }: CommandContext<AnyPluginData<any>>): TextChannel | ThreadChannel {
     const channelId = getChannelId(value);
     if (!channelId) {
       throw new TypeConversionError(`\`${disableCodeBlocks(value)}\` is not a valid channel`);
     }
 
-    if (!(message.channel instanceof GuildChannel)) {
+    if (message.channel.type === "DM") {
       throw new TypeConversionError(`Type 'Channel' can only be used in guilds`);
     }
```

Comment:
I believe this should be called before calling `getChannelId` for performance reasons

Reply:
True